tests/sanity/tests/contacts.spec.ts

```diff
@@ -1,142 +1,107 @@
-import { expect, test } from '@playwright/test'
+import { test } from '@playwright/test'
 import { fillSearch, generateId, PlatformSetting, PlatformURI } from './utils'
+import { ContractPage, ButtonAction } from './model/contacts/contract-page'
 
 test.use({
   storageState: PlatformSetting
 })
 
 test.describe('contact tests', () => {
+  let contractPage: ContractPage
+  let first: string
+  let last: string
+  let mail: string
+  let orgName: string
   test.beforeEach(async ({ page }) => {
+    first = 'Elton-' + generateId(5)
+    last = 'John-' + generateId(5)
+    mail = 'eltonjohn@' + generateId(5)
+    orgName = 'Company' + generateId(5)
+    contractPage = new ContractPage(page)
     await (await page.goto(`${PlatformURI}/workbench/sanity-ws`))?.finished()
   })
 
-  test('create-contact', async ({ page }) => {
-    // Create a new context with the saved storage state.
-    await page.locator('[id="app-contact\\:string\\:Contacts"]').click()
-
-    await page.click('.antiNav-element:has-text("Person")')
-    await page.click('button:has-text("Person")')
-
-    const first = 'Elton-' + generateId(5)
-    const last = 'John-' + generateId(5)
-
-    const firstName = page.locator('[placeholder="First name"]')
-    await firstName.click()
-    await firstName.fill(first)
-
-    const lastName = page.locator('[placeholder="Last name"]')
-    await lastName.click()
-    await lastName.fill(last)
-
-    await page.locator('.antiCard').locator('button:has-text("Create")').click()
-    await page.waitForSelector('form.antiCard', { state: 'detached' })
+  test('create-contact', async () => {
+    await contractPage.clickAppContact()
+    await contractPage.clickEmployeeNavElement('Person')
+    await contractPage.clickEmployeeButton('Person')
+    await contractPage.clickFirstNameInput()
```

Comment:
Why we need click and fill and not one operations? It is ok for recording, but for manual code it not looks good.

Reply:
we can either create a function that separately clicks and fills the input field, I believe that I presented both ways in this PR, or just fill the input field. Basically the purpose it is to mimic the users behavior, plus as you can see in the previous code it was done the same way
```
    const lastName = page.locator('[placeholder="Last name"]')
    await lastName.click()
    await lastName.fill(last)
```
, I didn't want to go full out of the way of how you have written the code. Before naturally mimicking the specific behaviour we would use .click() and then .type() which has been depricated in newer version of playwright

Reply:
Also, my current knowledge of the app and how it fully functions is quite limited so I didn't want to go full out of the way until I have a bigger grasp of everything. The funny thing from my previous experience was once on the project where tests were running on the CI we were not able to fill the locator unless the input field was clicked. I am not quite sure what is the state here, so as said I didn't want to go full-out mode with solutions that might not work on the separate environment. You probably have heard the saying "It works on my end locally"